feature/metis/conversation/src/test/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/service/storage/impl/MetisStorageServiceImplTest.kt

```diff
@@ -0,0 +1,165 @@
+import de.tum.informatics.www1.artemis.native_app.feature.metis.shared.content.dto.UserRole
+import de.tum.informatics.www1.artemis.native_app.feature.metis.shared.content.dto.conversation.OneToOneChat
+import de.tum.informatics.www1.artemis.native_app.feature.metis.shared.db.pojo.AnswerPostPojo
+import de.tum.informatics.www1.artemis.native_app.feature.metis.shared.db.pojo.PostPojo
+import de.tum.informatics.www1.artemis.native_app.feature.metistest.MetisDatabaseProviderMock
+import kotlinx.coroutines.test.runTest
+import kotlinx.datetime.Clock
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.experimental.categories.Category
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@Category(UnitTest::class)
+@RunWith(RobolectricTestRunner::class)
+class MetisStorageServiceImplTest {
+
+    private val databaseProviderMock = MetisDatabaseProviderMock(InstrumentationRegistry.getInstrumentation().context)
+    private val sut = MetisStorageServiceImpl(databaseProviderMock)
+
+    private val host = "host"
+
+    private val author = User(id = 20, name = "AuthorName")
+    private val parentClientPostId = "parent-client-id-0"
+    private val answerClientPostId = "answer-client-id-0"
+
+    private val course: MetisContext.Course = MetisContext.Course(courseId = 1)
+    private val conversation = OneToOneChat(id = 2)
+    private val metisContext = MetisContext.Conversation(course.courseId, conversation.id)
+
+    private val localAnswerPojo = AnswerPostPojo(
+        parentPostId = parentClientPostId,
+        postId = answerClientPostId,
+        resolvesPost = false,
+        basePostingCache = AnswerPostPojo.BasePostingCache(
+            serverPostId = 0,
+            authorId = author.id,
+            creationDate = Clock.System.now(),
+            updatedDate = null,
+            content = "Answer post content 0",
+            authorRole = UserRole.USER,
+            authorName = author.name!!
+        ),
+        reactions = emptyList(),
+        serverPostIdCache = AnswerPostPojo.ServerPostIdCache(
+            serverPostId = null     // Only local answer post, no server id
+        )
+    )
+
+    private val basePostPojo = PostPojo(
+        clientPostId = parentClientPostId,
+        serverPostId = 0,
+        content = "Base post content",
+        resolved = false,
+        updatedDate = null,
+        creationDate = Clock.System.now(),
+        authorId = author.id,
+        title = null,
+        authorName = author.name!!,
+        authorRole = UserRole.USER,
+        courseWideContext = null,
+        tags = emptyList(),
+        answers = emptyList(),
+        reactions = emptyList()
+    )
+
+    private val basePost = StandalonePost(basePostPojo, conversation)
+    private val localAnswer = AnswerPost(localAnswerPojo, basePost)
+
+    @Test
+    fun testInsertClientSidePost() = runTest {
```

Comment:
Please add the test GIVEN WHEN THEN naming as a comment to the test functions if it does not work with the method names.

Reply:
As discussed in [this thread](https://github.com/ls1intum/artemis-android/pull/96#discussion_r1835414778), this leads to error in the SQL database, thats why I opted to go for the shorter method naming and instead add the more descriptive ones in the comments of the test